src/components/UpdatePaymentDetailsPopup/Steps/Error.js

```diff
@@ -21,24 +37,26 @@ const Error = ({ hideInnerPopup }) => {
         </ImageWrapper>
         <TitleStyled>{t('Oops, something went wrong')}</TitleStyled>
         <TextStyled>
-          {t('Your payment details have not been updated.')}
+          {error?.includes('Refused') ? (
+            <Trans i18nKey="update-payment-error-method">
+              We weren’t able to update your payment method. <br /> Please try
+              again.
+            </Trans>
+          ) : (
+            <Trans i18nKey="update-payment-error-details">
+              We weren’t able to update your payment details. <br /> Please try
+              again using different payment method.
+            </Trans>
+          )}
         </TextStyled>
       </ContentStyled>
       <ButtonWrapperStyled removeMargin>
-        <Button theme="simple" onClickFn={() => hideInnerPopup()}>
-          {t('Back to Payment Details')}
+        <Button theme="simple" onClickFn={handleTryAgain}>
+          {t('Try again')}
```

Comment:
did you test the default case when this popup was presented? Eg. try to add a regular card as a payment method and provide wrong CVC. The copy will be updated to the new one and also button will be replaced but I think it's a better behavior then what we had previously

Reply:
I agree that current behavior is better. But thanks to that I found one bug - when error action is undefined which cause error ;)